docs/csharp/whats-new/csharp-version-history.md

```diff
@@ -250,4 +242,60 @@ C# 8.0 is the first major C# release that specifically targets .NET Core. Some f
 
+## C# version 9.0
+
+C# 9.0 was released with .NET 5. It's the default language version for any assembly that targets the .NET 5 release. It contains the following new and enhanced features:
+
+C# 9.0 adds the following features and enhancements to the C# language:
+
+- [Records](./csharp-9.md#record-types)
+- [Init only setters](./csharp-9.md#init-only-setters)
+- [Top-level statements](./csharp-9.md#top-level-statements)
+- [Pattern matching enhancements](./csharp-9.md#pattern-matching-enhancements)
+- [Performance and interop](./csharp-9.md#performance-and-interop)
+  - [Native sized integers](~/_csharplang/proposals/csharp-9.0/native-integers.md)
+  - [Function pointers](~/_csharplang/proposals/csharp-9.0/function-pointers.md)
+  - [Suppress emitting localsinit flag](~/_csharplang/proposals/csharp-9.0/skip-localsinit.md)
+- [Fit and finish features](./csharp-9.md#fit-and-finish-features)
+  - [Target-typed `new` expressions](~/_csharplang/proposals/csharp-9.0/target-typed-new.md)
+  - [`static` anonymous functions](~/_csharplang/proposals/csharp-9.0/static-anonymous-functions.md)
+  - [Target-typed conditional expressions](~/_csharplang/proposals/csharp-9.0/target-typed-conditional-expression.md)
+  - [Covariant return types](~/_csharplang/proposals/csharp-9.0/covariant-returns.md)
+  - [Extension `GetEnumerator` support for `foreach` loops](~/_csharplang/proposals/csharp-9.0/extension-getenumerator.md)
+  - [Lambda discard parameters](~/_csharplang/proposals/csharp-9.0/lambda-discard-parameters.md)
+  - [Attributes on local functions](~/_csharplang/proposals/csharp-9.0/local-function-attributes.md)
+- [Support for code generators](./csharp-9.md#support-for-code-generators)
+  - [Module initializers](~/_csharplang/proposals/csharp-9.0/module-initializers.md)
+  - [New features for partial methods](~/_csharplang/proposals/csharp-9.0/extending-partial-methods.md)
+
+C# 9.0 continues three of the themes from previous releases: removing ceremony, separating data from algorithms, and providing more patterns in more places.
+
+[Top level statements](../fundamentals/program-structure/top-level-statements.md) means your main program is simpler to read. There's less need for ceremony: a namespace, a `Program` class, and `static void Main()` are all unnecessary.
+
+The introduction of [`records`](../language-reference/builtin-types/record.md) provide a concise syntax for reference types that follow value semantics for equality. You'll use these types to define data containers that typically define minimal behavior. [Init-only setters](./csharp-9.md#init-only-setters) provide the capability for non-destructive mutation (`with` expressions) in records. C# 9.0 also adds [covariant return types](~/_csharplang/proposals/csharp-9.0/covariant-returns.md) so that derived records can override virtual methods and return a type derived from the base method's return type.
```

Comment:
is `type` I added here appropriate? Unsure.

```suggestion
The introduction of [`records`](../language-reference/builtin-types/record.md) provide a concise syntax for reference types that follow value type semantics for equality. You'll use these types to define data containers that typically define minimal behavior. [Init-only setters](./csharp-9.md#init-only-setters) provide the capability for non-destructive mutation (`with` expressions) in records. C# 9.0 also adds [covariant return types](~/_csharplang/proposals/csharp-9.0/covariant-returns.md) so that derived records can override virtual methods and return a type derived from the base method's return type.
```

Reply:
It's not bad, but "value semantics" is more correct.